v0/destinations/fbpixel/transform.js

```diff
@@ -0,0 +1,431 @@
+    const mappedTo = mapped.reduce((filtered, map) => {
+      if (map.from === category) {
+        filtered = map.to;
+      }
+      return filtered;
+    }, "");
+    if (mappedTo.length) {
+      return mappedTo;
+    }
+  }
+  return defaultValue;
+}
+
+function handleOrder(message, categoryToContent) {
+  const { products } = message.properties;
+  const value = formatRevenue(message.properties.revenue);
+  const contentType = getContentType(message, "product", categoryToContent);
+  const contentIds = [];
+  const contents = [];
+  const { category } = message.properties;
+
+  for (let i = 0; i < products.length; i += 1) {
+    const pId = products[i].product_id || products[i].sku || products[i].id;
+    contentIds.push(pId);
+    const content = {
+      id: pId,
+      quantity: products[i].quantity,
+      item_price: products[i].price
+    };
+    contents.push(content);
+  }
+  return {
+    content_category: category,
+    content_ids: contentIds,
+    content_type: contentType,
+    currency: message.properties.currency,
+    value,
+    contents,
+    num_items: contentIds.length
+  };
+}
+function handleProductListViewed(message, categoryToContent) {
+  let contentType;
+  const contentIds = [];
+  const contents = [];
+  const { products } = message.properties;
+  if (Array.isArray(products)) {
+    products.forEach(function(product) {
+      const productId = product.product_id;
+      if (productId) {
+        contentIds.push(productId);
+        contents.push({
+          id: productId,
+          quantity: message.properties.quantity
+        });
+      }
+    });
+  }
+
+  if (contentIds.length) {
+    contentType = "product";
+  } else {
+    contentIds.push(message.properties.category || "");
+    contents.push({
+      id: message.properties.category || "",
+      quantity: 1
+    });
+    contentType = "product_group";
+  }
+  return {
+    content_ids: contentIds,
+    content_type: getContentType(message, contentType, categoryToContent),
+    contents
+  };
+}
+
+function handleProduct(message, categoryToContent, valueFieldIdentifier) {
+  const useValue = valueFieldIdentifier === "properties.value";
+  const contentIds = [
+    message.properties.product_id ||
+      message.properties.id ||
+      message.properties.sku ||
+      ""
+  ];
+  const contentType = getContentType(message, "product", categoryToContent);
+  const contentName =
+    message.properties.product_name || message.properties.name || "";
+  const contentCategory = message.properties.category || "";
+  const { currency } = message.properties;
+  const value = useValue
+    ? formatRevenue(message.properties.value)
+    : formatRevenue(message.properties.price);
+  const contents = [
+    {
+      id:
+        message.properties.product_id ||
+        message.properties.id ||
+        message.properties.sku ||
+        "",
+      quantity: message.properties.quantity,
+      item_price: message.properties.price
```

Comment:
should this be different that value as derived earlier?

Reply:
it should be only set if price is present. value is dependent on the config